provider/pkg/provider/helm_release.go

```diff
@@ -432,6 +422,113 @@ func (r *helmReleaseProvider) setDefaults(target resource.PropertyMap) {
 	}
 }
 
+// setComputedInputs augments resource inputs with computed inputs.
+func (r *helmReleaseProvider) setComputedInputs(ctx context.Context, urn resource.URN, new *Release) error {
+
+	// generate the template output
+	manifest, err := r.helmTemplate(ctx, urn, new)
+	if err != nil {
+		return err
+	}
+
+	// compute the checksum of the rendered output, to be able to detect changes
+	h := sha256.New()
+	_, err = h.Write([]byte(manifest))
+	if err != nil {
+		return err
+	}
+	new.Checksum = hex.EncodeToString(h.Sum(nil))
+
+	// compute resource names
+	_, resources, err := convertYAMLManifestToJSON(manifest)
+	if err != nil {
+		return err
+	}
+	new.ResourceNames = resources
+
+	return nil
+}
+
+func (r *helmReleaseProvider) helmTemplate(ctx context.Context, urn resource.URN, newRelease *Release) (string, error) {
+	conf, err := r.getActionConfig(newRelease.Namespace)
+	if err != nil {
+		return "", err
+	}
+	client := action.NewInstall(conf)
+	c, path, err := getChart(&client.ChartPathOptions, conf.RegistryClient, r.settings, newRelease)
+	if err != nil {
+		logger.V(9).Infof("getChart failed: %+v", err)
+		logger.V(9).Infof("Settings: %#v", r.settings)
+		return "", err
+	}
+
+	logger.V(9).Infof("Checking chart dependencies for chart: %q with path: %q", newRelease.Chart, path)
+	// check and update the chart's dependencies if needed
+	updated, err := checkChartDependencies(
+		c,
+		path,
+		newRelease.Keyring,
+		r.settings,
+		conf.RegistryClient,
+		newRelease.DependencyUpdate)
+	if err != nil {
+		return "", err
+	} else if updated {
+		// load the chart again if its dependencies have been updated
+		c, err = loader.Load(path)
+		if err != nil {
+			return "", err
+		}
+	}
+
+	logger.V(9).Infof("Fetching values for release: %q", newRelease.Name)
+	values, err := getValues(newRelease)
+	if err != nil {
+		return "", err
+	}
+
+	logger.V(9).Infof("Values: %+v", values)
+
+	err = isChartInstallable(c)
+	if err != nil {
+		return "", err
+	}
+
+	client.DryRun = true
+	client.ClientOnly = true
+	client.Devel = newRelease.Devel
+	client.DependencyUpdate = newRelease.DependencyUpdate
+	client.Namespace = newRelease.Namespace
+	client.ReleaseName = newRelease.Name
+	client.GenerateName = false
+	client.NameTemplate = ""
+	client.OutputDir = ""
+	client.SubNotes = newRelease.RenderSubchartNotes
+	client.DisableOpenAPIValidation = true
+	client.Replace = true
+	client.APIVersions = nil
+	client.IncludeCRDs = !newRelease.SkipCrds
+
+	if cmd := newRelease.Postrender; cmd != "" {
+		pr, err := postrender.NewExec(cmd)
+
+		if err != nil {
+			return "", err
+		}
+
+		client.PostRenderer = pr
+	}
+
+	logger.V(9).Infof("template helm chart")
+	rel, err := client.RunWithContext(r.canceler.context, c, values)
+	if err != nil {
+		return "", err
+	}
+
+	manifest := getManifest(rel, !newRelease.DisableWebhooks, true)
+	return manifest, err
```

Comment:
nit: err is always nil here, can we return nil instead of err just to make that clearer?